Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreStepTaskContext.cs

```diff
@@ -0,0 +1,65 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.BlockStore;
+using Stratis.Bitcoin.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
+{
+    public sealed class BlockStoreStepTaskContext
+    {
+        public BlockStoreStepTaskContext(CancellationToken cancellationToken, BlockStoreLoop blockStoreLoop)
+        {
+            Guard.NotNull(blockStoreLoop, "blockStoreLoop");
+
+            this.BlockStoreLoop = blockStoreLoop;
+            this.CancellationToken = cancellationToken;
+        }
+
+        public BlockStoreStepTaskContext Initialize(ChainedBlock nextChainedBlock)
+        {
+            Guard.NotNull(nextChainedBlock, "nextChainedBlock");
+
+            this.DownloadStack = new Queue<ChainedBlock>(new[] { nextChainedBlock });
+            this.NextChainedBlock = nextChainedBlock;
+            this.Routine = new List<BlockStoreStepTask>() { new BlockStoreStepFindBlocksTask(), new BlockStoreStepDownloadBlocksTask() };
+
+            this.InsertBlockSize = 0;
+            this.StallCount = 0;
+            this.Store = new List<BlockPair>();
+
+            this.BlockStoreLoop.BlockPuller.AskBlock(nextChainedBlock);
```

Comment:
I would probably take out AskBlocks from context, context should not have much logic (especially not download logic) what do you think?